src/cloudevents/core/v1/event.py

```diff
@@ -0,0 +1,140 @@
+        """
+        Private method that validates the attributes of the CloudEvent as per the CloudEvents specification.
+        """
+        missing_attributes = [
+            attr for attr in REQUIRED_ATTRIBUTES if attr not in attributes
+        ]
+        if missing_attributes:
+            raise ValueError(
+                f"Missing required attribute(s): {', '.join(missing_attributes)}"
+            )
+
+        if attributes["id"] is None:
+            raise ValueError("Attribute 'id' must not be None")
+
+        if not isinstance(attributes["id"], str):
+            raise TypeError("Attribute 'id' must be a string")
+
+        if not isinstance(attributes["source"], str):
+            raise TypeError("Attribute 'source' must be a string")
+
+        if not isinstance(attributes["type"], str):
+            raise TypeError("Attribute 'type' must be a string")
+
+        if not isinstance(attributes["specversion"], str):
+            raise TypeError("Attribute 'specversion' must be a string")
+
+        if attributes["specversion"] != "1.0":
+            raise ValueError("Attribute 'specversion' must be '1.0'")
+
+        if "time" in attributes:
+            if not isinstance(attributes["time"], datetime):
+                raise TypeError("Attribute 'time' must be a datetime object")
+
+            if not attributes["time"].tzinfo:
+                raise ValueError("Attribute 'time' must be timezone aware")
+
+        if "subject" in attributes:
+            if not isinstance(attributes["subject"], str):
+                raise TypeError("Attribute 'subject' must be a string")
+
+            if not attributes["subject"]:
+                raise ValueError("Attribute 'subject' must not be empty")
+
+        if "datacontenttype" in attributes:
+            if not isinstance(attributes["datacontenttype"], str):
+                raise TypeError("Attribute 'datacontenttype' must be a string")
+
+            if not attributes["datacontenttype"]:
+                raise ValueError("Attribute 'datacontenttype' must not be empty")
+
+        if "dataschema" in attributes:
+            if not isinstance(attributes["dataschema"], str):
+                raise TypeError("Attribute 'dataschema' must be a string")
+
+            if not attributes["dataschema"]:
+                raise ValueError("Attribute 'dataschema' must not be empty")
+
+        for custom_extension in (
+            set(attributes.keys()) - REQUIRED_ATTRIBUTES - OPTIONAL_ATTRIBUTES
+        ):
+            if custom_extension == "data":
+                raise ValueError(
+                    "Extension attribute 'data' is reserved and must not be used"
+                )
+
+            if not custom_extension[0].isalpha():
+                raise ValueError(
+                    f"Extension attribute '{custom_extension}' should start with a letter"
+                )
+
+            if not (5 <= len(custom_extension) <= 20):
```

Comment:
as per the [spec ](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#naming-conventions): `MUST be at least one character in length, and SHOULD NOT exceed 20 characters.`

Reply:
Ah, don't know where I got the `5`. Good catch